### ISSUE TITLE
track warning for `mgl-markers-for-clusters`

### ISSUE THREAD

**Issue by HarelM** [maplibre/ngx-maplibre-gl#184]
When there's no `feature.id` with latest version of angular, there's a warning to the console about the fact that track by is not unique:
```
NG0955: The provided track expression resulted in duplicated keys for a given collection. Adjust the tracking expression such that it uniquely identifies all the items in the collection. Duplicated keys were: 
key "" at index "0" and "1", 
key "" at index "1" and "2", 
key "" at index "2" and "3", 
key "" at index "3" and "4", 
key "" at index "4" and "5". Find more at https://angular.dev/errors/NG0955
```

This is coming from the following line:
https://github.com/maplibre/ngx-maplibre-gl/blob/08949de5f6b57b13e7853ee2e1d50f50c32637a3/projects/ngx-maplibre-gl/src/lib/markers-for-clusters/markers-for-clusters.component.ts#L83

I'm not sure what should be the solution to this, but I think it would be great to solve this before the next release.
CC: @marcjulian

**Comment by marcjulian** [comment]
We could use `track $index` instead of `feature.id`

**Comment by HarelM** [comment]
That was my thought as well reading the following SO question:
https://stackoverflow.com/questions/77447092/angular-ng5002-for-loop-must-have-a-track-expression

**Comment by marcjulian** [comment]
Yes think so it would be better to use $index as feature.id may be undefined.
